### PR TITLE
Fix config prop parsing in Script::Commands::Enable

### PR DESCRIPTION
**Pull request by jacobsteves**
### WHY are these changes introduced?

Closes https://github.com/Shopify/script-service/issues/1757

Enabling a script with specific `config_props` currently has a couple of bugs:
1. `key1:a, key2:b` is parsed as `{"key1": "a", " key2": "b"}` (`key2` has an extra space in the key).
2. `key1:a:invalid_action` raises an ugly error

### WHAT is this pull request doing?

Fixes the two bugs above by validating the config_props. If invalid, it raises the following error:

image

@kwringe please let me know if you have a better error message 🙂

### CODE REVIEW

File: test/project_types/script/commands/enable_test.rb [hunk 1]
```diff
@@ -112,34 +92,45 @@ def test_calls_application_enable_with_configuration_properties
...
+
+        capture_io do
+          perform_command(config_props: "key1:value1 , key2:with spaces")
+        end
+      end
+
+      def test_should_call_error_handler_when_given_invalid_config_props
+        Script::UI::ErrorHandler
+          .expects(:pretty_print_and_raise)
+          .with { |e| e.is_a?(Errors::InvalidConfigProps) }
```
Comment (andyw8) [inline test/project_types/script/commands/enable_test.rb:127]: Maybe use the `instance_of` matcher here?

https://mocha.jamesmead.org/Mocha/ParameterMatchers.html#instance_of-instance_method

**Review by ErinRenNumber1 — APPROVED** [review]
LGTM aside from what Andy is suggesting!!  👍

**Comment by kwringe** [comment]
How about:

Can't enable script because config-props is formatted incorrectly. Try again using this format --config....

File: test/project_types/script/commands/enable_test.rb (hunk 1 above)
Comment (jacobsteves) [reply test/project_types/script/commands/enable_test.rb]: TIL, I like that. Updated it!

**Comment by jacobsteves** [comment]
Talked with @kwringe on slack and we settled on `Can't enable script. --config_props is formatted incorrectly. Try again using this format: --config_props='name1:value1, name2:value2'.` to be consistent with the other script command errors.

**Pull request merged**